### DISCUSSION TITLE
Static Nat support

### DISCUSSION THREAD

**Discussion by redpoisonv3** [fabianishere/udm-kernel-tools#68] [General]
i was wondering if static nat requires specific con track modules or an update to ip tables in order for it to be supported?
when io asked about full cone nat before i meant static nat sorry for the prior confusion.

**Comment by fabianishere** [comment]
I honestly don’t know. However, I don’t expect you need a custom kernel to do static NAT (just some custom kernel modules perhaps).

**Comment by redpoisonv3** [comment]
sorry i meant to ask for an updated minupnp, and mini ssdp version since the current one used is dated to 2018, since i can't use the rules for iptables without the newer miniupnp components.

**Reply by fabianishere** [reply]
Are these part of the kernel? Could you tell me which config options you are looking for?

**Reply by redpoisonv3** [reply]
im assuming since, the current version is coded to this version, keep in mind i could be wrong.
ii  minissdpd                            1.5.20180223-6+ubnt-65+g6366335     arm64        keep memory of all UPnP devices that announced themselves
ii  miniupnpd                            2.1-6.1+ubnt-113+g9236ffe           arm64        UPnP and NAT-PMP daemon for gateway routers

those are the associated modules, that are currently bundled.

port control protocol or Nat-PCP is available with the newer version it replaces nat pmp, there should be an outbound port config option that allows for the outbound and inbound ports to match when a upnp request is made, because currently its set to randomise the outbound port.

**Reply by redpoisonv3** [reply]
here is a link to minu upnp https://github.com/miniupnp/miniupnp

**Reply by fabianishere** [reply]
From what I can tell, that’s a user space daemon, so you don’t need a custom kernel for that.

**Reply by redpoisonv3** [reply]
i checked thanks you were right, one other question in the custom firmware does cake replace fq_codel , and what is the level of customisation available?

**Reply by fabianishere** [reply]
CAKE is available, but you have to configure it to replace fq_codel manually. It's just Linux kernel, so you can play with any of the kernel options available.
